File: test_sofa.py
import requests
import json

headers = {
    'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Safari/537.36',
    'Accept': 'application/json, text/plain, */*',
    'Origin': 'https://www.sofascore.com',
    'Referer': 'https://www.sofascore.com/',
}

url = 'https://api.sofascore.com/api/v1/team/4481/players'
response = requests.get(url, headers=headers)
print(f"Status: {response.status_code}")
try:
    data = response.json()
    print("Keys in response:", data.keys())
    if 'players' in data:
        print(f"Number of players: {len(data['players'])}")
        if len(data['players']) > 0:
            print("First player keys:", data['players'][0].keys())
            if 'player' in data['players'][0]:
                print("First player inner keys:", data['players'][0]['player'].keys())
except Exception as e:
    print("Error parsing JSON:", e)
    print("Response text:", response.text[:200])

File: flutter_application_1/test_sofa.dart
import 'package:http/http.dart' as http;

void main() async {
  final url = Uri.parse(
    'https://api.sofascore.com/api/v1/unique-tournament/16/season/58210/events/round/1',
  );
  final headers = {
    'Accept': 'application/json, text/plain, */*',
    'Accept-Language': 'fr-FR,fr;q=0.9',
    'User-Agent':
        'Mozilla/5.0 (Linux; Android 13; Infinix) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/124.0.0.0 Mobile Safari/537.36',
    'Referer': 'https://www.sofascore.com/',
    'Origin': 'https://www.sofascore.com',
  };

  try {
    print('Fetching from $url...');
    final res = await http.get(url, headers: headers);
    print('Status: ${res.statusCode}');
    if (res.statusCode == 200) {
      print('Success: ${res.body.substring(0, 100)}');
    } else {
      print('Error Body: ${res.body}');
    }
  } catch (e) {
    print('Exception: $e');
  }
}
